src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -27,7 +27,8 @@ public class AzureAppConfigurationOptions
         private List<IKeyValueAdapter> _adapters = new List<IKeyValueAdapter>() 
         { 
             new AzureKeyVaultKeyValueAdapter(new AzureKeyVaultSecretProvider()),
-            new JsonKeyValueAdapter() 
+            new JsonKeyValueAdapter(),
+            new FeatureManagementKeyValueAdapter()
```

Comment:
Why are we adding the feature management adapter by default? This would mean that we are flattening feature flags even when `UseFeatureFlags` is not specified.

Reply:
There was an impromptu discussion about this. the conclusion was that we need to handle feature flags that are in snapshots, even if UseFeatureFlags isn't called. The proposal was to add the feature flag adapater by default similarly to json and key vault adapters.

Reply:
That's what I thought, but we should call this out as a potential breaking change for users who might be expecting raw feature flags in their application.

Reply:
Sounds like a major version bump is needed for select snapshot.